src/node/internal/internal_http_incoming.ts

```diff
@@ -60,86 +94,335 @@ export class IncomingMessage extends Readable implements _IncomingMessage {
     });
 
     this.on('timeout', () => {
-      this.#reading = false;
+      this._consuming = false;
     });
 
-    this.#reader = this.#response.body?.getReader();
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.#tryRead();
+    this._stream = this.#response.body;
   }
 
   async #tryRead(): Promise<void> {
-    if (!this.#reader || this.#reading || this.#aborted) return;
-
-    this.#reading = true;
+    if (this._stream == null) return;
 
+    this.#reader ??= this._stream.getReader();
     try {
-      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-      while (true) {
-        const { done, value } = await this.#reader.read();
-        if (done) {
-          break;
-        }
-        this.push(value);
+      const data = await this.#reader.read();
+      if (data.done) {
+        // Done with stream, tell Readable we have no more data;
+        this.complete = true;
+        this.push(null);
+      } else {
+        this.push(data.value);
       }
-    } catch (error) {
-      this.emit('error', error);
-    } finally {
-      this.#reading = false;
-      this.#resetTimers({ finished: true });
+    } catch (e) {
+      this.destroy(e as Error);
+    }
+  }
+
+  // As this is an implementation of stream.Readable, we provide a _read()
+  // function that pumps the next chunk out of the underlying ReadableStream.
+  override _read(_n: number): void {
+    if (!this._consuming) {
+      if (this._readableState) {
+        this._readableState.readingMore = false;
+      }
+      this._consuming = true;
+    }
+
+    // Difference from Node.js -
+    // The Node.js implementation will already have its internal buffer
+    // filled by the parserOnBody function.
+    // For our implementation, we use the ReadableStream instance.
+    if (this._stream == null) {
+      // For GET and HEAD requests, the stream would be empty.
+      // Simply signal that we're done.
+      this.complete = true;
       this.push(null);
+      return;
     }
+
+    this.#tryRead().catch(() => {
+      /* Ignore errors */
```

Comment:
Why ignore errors?

Reply:
Without this catch, eslint errors. I'll just ignore this eslint error here.

```
143:5  error  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
```